patroni/postgresql/__init__.py

```diff
@@ -1031,13 +1042,14 @@ def restart(self, timeout: Optional[float] = None, task: Optional[CriticalTask]
 
         :returns: True when restart was successful and timeout did not expire when waiting.
         """
-        self.set_state('restarting')
+        self.set_state(PostgresqlState.RESTARTING)
         if not block_callbacks:
             self.__cb_pending = CallbackAction.ON_RESTART
         ret = self.stop(block_callbacks=True, before_shutdown=before_shutdown)\
             and self.start(timeout, task, True, role, after_start)
         if not ret and not self.is_starting():
-            self.set_state('restart failed ({0})'.format(self.state))
+            self.set_state(PostgresqlState.RESTART_FAILED)
+            logger.warning('restart failed (%r)', self.state)
```

Comment:
We want to write logs _before_ changing the state.
```suggestion
            logger.warning('restart failed (%r)', self.state)
            self.set_state(PostgresqlState.RESTART_FAILED)
```